### ISSUE TITLE
Macos unstable packages

### ISSUE THREAD

**Issue by matschaffer-roblox** [fluent/fluent-bit#6531]
Continuing the conversation with @patrick-stephens from https://github.com/fluent/fluent-bit-docs/pull/995#issuecomment-1342410966

Looks like there should be a mac binary produced from

- https://github.com/fluent/fluent-bit/suites/9739999362/artifacts/467527356
- https://github.com/fluent/fluent-bit/actions/runs/3645742948

But they're not appearing on https://fluentbit-staging.s3.amazonaws.com/

**Comment by matschaffer-roblox** [comment]
The run on https://github.com/fluent/fluent-bit/actions/runs/3645742948/jobs/6156524273 shows the step got skipped. Possibly due to the `if: inputs.environment == 'staging'` check?

**Comment by matschaffer-roblox** [comment]
Yeah, I can see it runs on https://github.com/fluent/fluent-bit/actions/runs/3515729290/jobs/5891577742 but I can't see the bucket name. Also it looks like it's only building an intel package where most of my team is on M1 now.

**Comment by matschaffer-roblox** [comment]
As best I can tell Github Actions doesn't yet support M1 runners. So we'll probably need a self-hosted runner or cross-compilation here.

**Comment by patrick-stephens** [comment]
Yeah we've been looking at options for M1 but it'll have to be self hosted - we actually discussed this in a community meeting with recently.

The bucket name should be: https://fluentbit-staging.s3.amazonaws.com/
Can you guess what the release one might ? :)

**Comment by patrick-stephens** [comment]
> The run on https://github.com/fluent/fluent-bit/actions/runs/3645742948/jobs/6156524273 shows the step got skipped. Possibly due to the `if: inputs.environment == 'staging'` check?

Feel free to PR this out so we can get it into staging and I'll run one manually for `master` to check. If you do the PR I can approve and merge.

**Comment by patrick-stephens** [comment]
The one thing I'd like before officially adding it to release targets is some testing on Macs - I'm afraid I have none available so it's a pain for me :(

**Comment by matschaffer-roblox** [comment]
Yeah, that's in interesting challenge. I'd offer the one in front of me but I'm pretty sure I'd need to get some sort of approval for that. 

@chenrui333 - it seems like you've been keeping the homebrew package up to date. Are you just building artifacts locally and publishing them?

**Comment by patrick-stephens** [comment]
It's probably just a case of getting people to try the staging ones to confirm they're ok.

**Comment by matschaffer** [comment]
That’ll be tough without aarch64 builds though since more and more people are moving off intel over time.

**Comment by patrick-stephens** [comment]
Yeah, I'm happy to have any suggestions for how to build M1 - we can set up something to do ephemeral workloads on self-hosted runners too but that's some work. Github recommends against using self-hosted runners in general for public repos (unknown PRs from forks are the issue I think) so I'd like to limit the amount of effort required to maintain this infrastructure.

The community meeting covered: https://docs.actuated.dev/
I've also been looking at other options around it.

The blocker is unknown demand really (captured here so thanks) and having time to sort it.

There is also this: https://github.com/github/roadmap/issues/528
However Mac runners currently have severe limitations anyway (they cost more and we can only use 5 concurrently which is consumed by a single unit test workflow run).

**Comment by patrick-stephens** [comment]
Can we do cross-compilation on the current runners?

**Comment by patrick-stephens** [comment]
My other thought was I could set up a private repo to build the source for Fluent Bit releases on a dedicated runner. It would not expose the build process (or runner) publicly then.

What do you think @niedbalski ? We could set up some M1 runners maybe with Equinix or others.

**Comment by matschaffer** [comment]
Here's a response I got via mastodon from https://mastodon.online/@flavorjones@ruby.social/109517824984135573

Might be worth trying https://github.com/tpoechtrager/osxcross for fluent-bit builds from linux workers.
